lib/manageiq/api/common/headers.rb

```diff
@@ -2,17 +2,30 @@ module ManageIQ
   module API
     module Common
       module Headers
+        FORWARDABLE_HEADER_KEYS = %w(X-Request-ID x-rh-identity).freeze
         def self.current=(val)
-          validate_headers(val)
+          validate_headers(val) if val
           Thread.current[:attr_current_headers] = val
```

Comment:
Should this be removed `if val == nil`?

Reply:
@bdunne  we are trying to unset a thread variable so we need to set it to a nil it to a nil and valdiate_headers rejects a nil because it is expecting a header object. So either the ensure block can directly set the thread variable to a nil or go thru current=